docs/contributing/devguide.md

```diff
@@ -47,7 +47,7 @@ Building the operator binary and docker image:
 
 ```shell
 make build
-make docker.build IMG=external-secrets:latest
+make docker.build DOCKER_BUILD_ARGS="--build-arg TARGETARCH=$(go env GOARCH) --build-arg TARGETOS=$(go env GOOS)"
```

Comment:
I've tried running `make docker.build` by itself and it didn't work, I've looked at the code and found out that when it's called from other build scripts those values are passed, I passed them and it worked.

I've followed the blames, the parameters passed changed across the years, thought it was just stale docs, if you can run `make docker.build` from a clean repo and it works let me know, I'll retry.

Reply:
`make docker.build` works fine in my local environment (M1 Mac, Docker version 24.0.2, Go version 1.19.3). Would you share the error message you received when you ran the command?

Reply:
```
$ LANG=C make docker.build 
09:13:24 [ OK ] Finished generating deepcopy and crds
09:13:24 [ .. ] go build amd64
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 \
	go build -o 'bin/external-secrets-linux-amd64' main.go
09:13:24 [ OK ] go build amd64
09:13:24 [ .. ] go build arm64
CGO_ENABLED=0 GOOS=linux GOARCH=arm64 \
	go build -o 'bin/external-secrets-linux-arm64' main.go
09:13:25 [ OK ] go build arm64
09:13:25 [ .. ] docker build
echo docker build -f Dockerfile .  -t ghcr.io/external-secrets/external-secrets:v0.9.1-17.g8c8c17e0
docker build -f Dockerfile . -t ghcr.io/external-secrets/external-secrets:v0.9.1-17.g8c8c17e0
docker build -f Dockerfile .  -t ghcr.io/external-secrets/external-secrets:v0.9.1-17.g8c8c17e0
Sending build context to Docker daemon  292.9MB
Step 1/6 : FROM gcr.io/distroless/static
 ---> 961207f62413
Step 2/6 : ARG TARGETOS
 ---> Using cache
 ---> dec1fd805da3
Step 3/6 : ARG TARGETARCH
 ---> Using cache
 ---> 3c50e8bb496e
Step 4/6 : COPY bin/external-secrets-${TARGETOS}-${TARGETARCH} /bin/external-secrets
COPY failed: file not found in build context or excluded by .dockerignore: stat bin/external-secrets--: file does not exist
make: *** [Makefile:243: docker.build] Error 1
```
As you see, it breaks because `${TARGETOS}` and `${TARGETARCH}` are empty, so I gave them.
You may have them set in your shell config.

Reply:
I briefly looked into it and noticed `TARGETOS ` and `TARGETARCH` are arguments automatically set by BuildKit. Would you check if you enabled it in your environment? 

- https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope
- https://docs.docker.com/build/buildkit/

Reply:
Mystery revealed, I'm on version 20.10, if I use `DOCKER_BUILDKIT=1 make docker.build` it works.
Had no idea they were changing the build backed.

Fixed, but I could also drop the cange altogether if desired.

Reply:
Fixed all around.